offchain-modules/packages/x/src/monitor/rpc-metric.ts

```diff
@@ -0,0 +1,45 @@
+import * as Prometheus from 'prom-client';
+import { logger } from '../utils/logger';
+
+export type responseStatus = 'success' | 'failed';
+
+export class RpcMetric {
+  public rpcRequestDurationms: Prometheus.Histogram<any>;
+  private gateway: Prometheus.Pushgateway;
+  private openPushMetric: boolean;
+  constructor(pushGatewayURL: string) {
+    const register = new Prometheus.Registry();
+    this.rpcRequestDurationms = new Prometheus.Histogram({
+      name: 'rpc_request_duration_ms',
+      help: 'Duration of rpc server requests in ms',
+      labelNames: ['method', 'status'],
+      buckets: [0.1, 5, 15, 50, 100], // buckets for response time from 0.1ms to 100ms
```

Comment:
IMHO, I'dont think the API like `generateXTransaction` could resolved in 0.1ms - 100ms, maybe we can use the [default buckets](https://github.com/siimon/prom-client/blob/c7b9a9ddd048f6a1921fa1b33c56a5296713aa2b/lib/histogram.js#L15) from prom-client recommended, which is `[0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10]`